doc/extending/using_params.txt

```diff
@@ -0,0 +1,140 @@
+.. _extending_op_params:
+
+===============
+Using Op params
+===============
+
+The Op params is a facility to pass some runtime parameters to the
+code of an op without modifying it.  It can enable a single instance
+of C code to serve different needs and therefore reduce compilation.
+
+I will first introduce the parts involved in actually using this
+functionality and then present a simple working example.
+
+The params type
+----------------
+
+You can either reuse an existing type such as :class:`Generic` or
+:class:`CDataType`, or create your own.
```

Comment:
I feel there should be a bit more information here to help a user reading the tutorial decide which of these 3 options is right for him/her.